templates.go

```diff
@@ -69,13 +71,22 @@ func initServerNameTemplates() (err error) {
 }
 
 type PortSpec struct {
+	IP       net.IP
 	Port     int32
 	Mode     string
 	Protocol string
 }
 
+// String representation of a PortSpec, intended to be used as config label
 func (s PortSpec) String() string {
-	return fmt.Sprintf("%d_%s_%s", s.Port, s.Protocol, s.Mode)
+	var encodedIP string
+	if s.IP.To4() != nil {
+		// No need to use more than 4 bytes if it is an IPv4 address
+		encodedIP = hex.EncodeToString(s.IP.To4())
+	} else {
+		encodedIP = hex.EncodeToString(s.IP)
+	}
+	return fmt.Sprintf("%s_%d_%s_%s", encodedIP, s.Port, s.Protocol, s.Mode)
```

Comment:
Previously, label had also the namespace included in this string. Is it not needed anymore ? 
I'm thinking of two services in different namespaces using the same port

Reply:
This label was used as is for the port (haproxy frontend) and didn't include the namespace, the main difference is that it includes now also a representation of the IP.

The label with a namespace is used when identifying services (haproxy backends) and it still contains the namespace, it is built now with the `String()` method of the Service to avoid code repetition in the templates.

Reply:
Ok, I see it now
I missed the String() method for ServiceInformation struct.